core/src/main/java/org/elasticsearch/index/query/NotQueryParser.java

```diff
@@ -56,12 +55,16 @@ public NotQueryBuilder fromXContent(QueryParseContext parseContext) throws IOExc
                 // skip
             } else if (token == XContentParser.Token.START_OBJECT) {
                 if (parseContext.parseFieldMatcher().match(currentFieldName, QUERY_FIELD)) {
-                    query = parseContext.parseInnerFilterToQueryBuilder();
+                    query = parseContext.parseInnerQueryBuilder();
                     queryFound = true;
                 } else {
                     queryFound = true;
                     // its the filter, and the name is the field
-                    query = parseContext.parseInnerFilterToQueryBuilder(currentFieldName);
+                    QueryParser queryParser = parseContext.queryParser(currentFieldName);
```

Comment:
might be something that was previously supported in and query and or query which are now gone. not is the only survivor. That said I am leaning towards leaving that logic in the context anyway? It feels weird to be looking queries up from the parser here and throwing generic exceptions like "no query registered". And maybe open an issue for discussion about deprecating this syntax. maybe we should have a single way to specify things here.